src/utils/format.ts

```diff
@@ -251,14 +253,15 @@ export function formatSmart(
     decimals = params.decimals ?? decimals
     smallLimit = params.smallLimit ?? smallLimit
     isLocaleAware = params.isLocaleAware ?? isLocaleAware
+    thousandSeparator = params.thousandSeparator === undefined ? thousandSeparator : params.thousandSeparator
```

Comment:
Isn't this the place to use the fancy `??` (look at the lines before this one)

```suggestion
    thousandSeparator = params.thousandSeparator ?? thousandSeparator
```

Reply:
does this allow for `false` or fallthrough?

Reply:
Will only use `thousandSeparator` if `params.thousandSeparator` is `null` or `undefined` => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator